harness/determined/pytorch/_reducer.py

```diff
@@ -19,25 +186,34 @@ class Reducer(Enum):
         MIN
     """
 
-    # TODO: Support additional reducers.
     AVG = 1
     SUM = 2
     MAX = 3
     MIN = 4
 
 
-def _reduce_metrics(
-    reducer: Reducer, metrics: np.array, num_batches: Optional[List[int]] = None
-) -> np.float:
-    if reducer == Reducer.AVG:
-        if num_batches:
-            check.check_eq(len(metrics), len(num_batches))
-        return np.average(metrics, weights=num_batches)
-    elif reducer == Reducer.SUM:
-        return np.sum(metrics)
-    elif reducer == Reducer.MAX:
-        return np.max(metrics)
-    elif reducer == Reducer.MIN:
-        return np.min(metrics)
-    else:
-        raise NotImplementedError
+def _make_reducer(r: ReducerType) -> MetricReducer:
+    if isinstance(r, Reducer):
```

Comment:
blocking: can you add a comment here "backward compatibility: support reducer in the old interface. This will be removed in ...."